apps/admin-dashboard/app/routes/_dashboard.applications.$id.accept.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  type ActionFunctionArgs,
+  json,
+  type LoaderFunctionArgs,
+  redirect,
+} from '@remix-run/node';
+import {
+  Form as RemixForm,
+  useActionData,
+  useLoaderData,
+  useNavigate,
+} from '@remix-run/react';
+
+import { Button, Form, Modal } from '@oyster/ui';
+
+import { Route } from '../shared/constants';
+import { acceptApplication, getApplication } from '../shared/core.server';
+import {
+  commitSession,
+  ensureUserAuthenticated,
+  toast,
+  user,
+} from '../shared/session.server';
+
+export async function loader({ params, request }: LoaderFunctionArgs) {
+  await ensureUserAuthenticated(request, {
+    allowAmbassador: true,
+  });
+
+  const application = await getApplication(params.id as string, [
+    'applications.firstName',
+    'applications.lastName',
+  ]);
+
+  if (!application) {
+    throw new Response(null, { status: 404 });
+  }
+
+  return json({
+    application,
+  });
+}
+
+export async function action({ params, request }: ActionFunctionArgs) {
+  const session = await ensureUserAuthenticated(request, {
+    allowAmbassador: true,
+  });
+
+  const adminId = user(session);
+
+  try {
+    await acceptApplication(params.id as string, adminId);
+
+    toast(session, {
+      message: 'Application has been accepted.',
+      type: 'success',
+    });
+
+    return redirect(Route.APPLICATIONS, {
+      headers: {
+        'Set-Cookie': await commitSession(session),
+      },
+    });
+  } catch (e) {
+    if ((e as Error).message.includes('FK_7b8cdcaabacb80df1ed8f7dbbea')) {
```

Comment:
Added this statement to provide some context when facing the error I received: 
> insert or update on table "applications" violates foreign key constraint "FK_7b8cdcaabacb80df1ed8f7dbbea"

There may be a better way to catch this error